src/effects/native/metronomeeffect.cpp

```diff
@@ -90,13 +91,15 @@ void MetronomeEffect::processChannel(
 
     unsigned int maxFrames;
     if (m_pSyncParameter->toBool() && groupFeatures.has_beat_length_sec) {
+        m_isBuffered = false;
         maxFrames = bufferParameters.sampleRate() * groupFeatures.beat_length_sec;
         if (groupFeatures.has_beat_fraction) {
             unsigned int currentFrame =  maxFrames * groupFeatures.beat_fraction;
             if (maxFrames > clickSize &&
-                    currentFrame > clickSize &&
-                    currentFrame < maxFrames - clickSize &&
-                    gs->m_framesSinceClickStart > clickSize) {
+                    currentFrame >= clickSize &&
+                    gs->m_framesSinceClickStart >= bufferParameters.framesPerBuffer() &&
+                    gs->m_framesSinceClickStart >= clickSize &&
+                    (!m_isBuffered || currentFrame < maxFrames - bufferParameters.framesPerBuffer())) {
```

Comment:
Isn't m_isBuffered always false here?

Reply:
The `m_isBuffered` bool would help in the case when the track was paused in between the beat. This will make sure that it does not keep playing the ticking sound continuously.